src/main/java/com/bunq/sdk/context/UserContext.java

```diff
@@ -18,54 +19,41 @@
   private static final String ERROR_NO_ACTIVE_MONETARY_ACCOUNT_FOUND = "No active monetary account found.";
   private static final String ERROR_PRIMARY_MONETARY_ACCOUNT_IS_NOT_SET = "Primary monetaryAccount is not set.";
 
-  private static final String MONETARY_ACCOUNT_STATUS_ACTIVE = "ACTIVE";
-  private static final int INDEX_FIRST = 0;
-
+  private final ApiContext apiContext;
   private UserCompany userCompany;
   private UserPerson userPerson;
   private MonetaryAccountBank primaryMonetaryAccountBank;
-  private Integer userId;
-
-  public UserContext(Integer userId) {
-    this.setUser(this.getUserObject());
-    this.userId = userId;
-  }
 
-  private BunqModel getUserObject() {
-    return User.list().getValue().get(INDEX_FIRST).getReferencedObject();
+  public UserContext(ApiContext apiContext) {
+    this.apiContext = apiContext;
+    refreshContext();
   }
 
-  private void setUser(BunqModel user) {
-    if (user instanceof UserPerson) {
-      this.userPerson = (UserPerson) user;
-    } else if (user instanceof UserCompany) {
-      this.userCompany = (UserCompany) user;
+  private void initUser(User user) {
+    if (user!=null && user.getUserPerson()!=null) {
```

Comment:
why are you not using `instaceof` anymore ?

Reply:
The original user var was based on User#getReferencedObject internally retrieved the BunqModel from either userPerson or userCompany, directly using the relevant accessors omitted the need to instanceof check

Reply:
This increases the complexity of the if statement  without any benefits. So please revert back to using `instaceof`

Reply:
removed the null-checks, the risk for nullpointers is the same as the original code, extra checking was merely defensive, it now has the same complexity

Reply:
btw; your code review thingie should mark the`instanceof`statement as some form of technical debt, it is not best practice and is generally avoidable (and is a code-smell)